### ISSUE TITLE
[Bug]: jsx-key could not be triggered when using a custom mapping function

### ISSUE THREAD

**Issue by meteorlxy** [jsx-eslint/eslint-plugin-react#3541]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

When using a custom array mapping function like `lodash.map`, the `jsx-key` rule won't be triggered

```tsx
import map from 'lodash/map';

// not triggered
const Comp1 = () => {
  return map([1, 2, 3], (item) => {
    return <div>{item}</div>;
  });
};

// triggered
const Comp2 = () => {
  return [1, 2, 3].map((item) => {
    return <div>{item}</div>;
  });
};
```

### Expected Behavior

Is it possible to add a config to support custom mapping function?

### eslint-plugin-react version

v7.32.2

### eslint version

v8.35.0

### node version

v18.14.0

**Comment by ljharb** [comment]
No, because there's no way to statically know what kind of function it is.

(Why would you use lodash map anyways when mapping functions have existed in the language for over a decade?)

**Comment by meteorlxy** [comment]
The project convention asks contibutors to always use lodash utils if possible. (I never use them like this in my personal projects 😅)

Thus, I found there are tons of FN of this rule in that project, and it seems that other contributors do not care about the runtime errors. (which is really annoying for me...)

If it's not possible for this eslint rule, I think I'd better ask for changing the project convention.

**Comment by ljharb** [comment]
I suspect that's the easiest approach, given that the vast majority of lodash is in the language now.
